lib/internal/streams/async_iterator.js

```diff
@@ -127,7 +127,10 @@ const createReadableStreamAsyncIterator = (stream) => {
     [kLastResolve]: { value: null, writable: true },
     [kLastReject]: { value: null, writable: true },
     [kError]: { value: null, writable: true },
-    [kEnded]: { value: false, writable: true },
+    [kEnded]: {
+      value: stream._readableState.endEmitted,
```

Comment:
I’m worried about reaching into stream internals. Could this be made publicly accessible?

Reply:
We can, but this instance is actually ok. We are in the streams internals here, I wouldn’t worry.